gene/fromGtf.go

```diff
@@ -58,6 +56,16 @@ func GtfToGene(g *gtf.Gene, ref []*fasta.Fasta) *Gene {
 					currCdsPos++
 				}
 			}
+
+			if val.ThreeUtr != nil {
+				for i = val.ThreeUtr.Start - 1 - answer.startPos; i < val.ThreeUtr.End-answer.startPos; i++ {
+					answer.featureArray[i] = UtrThree
+				}
+				if answer.utrThree.start == 0 {
```

Comment:
Again, not a bug, I think I am must having trouble telling what is happening.  I am not why/how a 3' UTR would have a start position of zero.  I guess there would be no 5' UTR and no coding seq.  Is this catching ncRNA genes, which are encoded as all 3' UTR?  I am also not sure why in the "if" we then set the 3' UTR to start the base after the cDNA ends; it seems like that would give an index-out-of-range when used.

Reply:
So this relies on the fact that GTF is organized such that the 5' UTR comes before the CDS comes before the 3' UTR. By the time a 3' UTR line is being read the 5' UTR and CDS sequences have already been recorded. This if statement says "if the 3' UTR start is unset (==0), then this must be the first 3'UTR sequence we encountered, therefore the start will be equal to wherever we currently are in the cDNA encoding." I added a couple comments to hopefully help clear this up.